adserver/views.py

```diff
@@ -956,6 +958,110 @@ def get_context_data(self, **kwargs):
         return context
 
 
+class AdvertiserAuthorizedUsersView(
+    AdvertiserAccessMixin, UserPassesTestMixin, ListView
+):
+
+    """Authorized users for an advertiser."""
+
+    model = get_user_model()
+    template_name = "adserver/advertiser/users.html"
+
+    def get_context_data(self, **kwargs):  # pylint: disable=arguments-differ
+        context = super().get_context_data(**kwargs)
+        context.update({"advertiser": self.advertiser, "users": self.object_list})
+        return context
+
+    def get_queryset(self):
+        self.advertiser = get_object_or_404(
+            Advertiser, slug=self.kwargs.get("advertiser_slug", "")
+        )
+        return self.advertiser.user_set.all()
+
+
+class AdvertiserAuthorizedUsersInviteView(
+    AdvertiserAccessMixin, UserPassesTestMixin, CreateView
+):
+
+    """Invite additional authorized users for an advertiser."""
+
+    form_class = InviteUserForm
+    model = get_user_model()
+    template_name = "adserver/advertiser/users-invite.html"
+
+    def dispatch(self, request, *args, **kwargs):
+        self.advertiser = get_object_or_404(
+            Advertiser, slug=self.kwargs["advertiser_slug"]
+        )
+        return super().dispatch(request, *args, **kwargs)
+
+    def form_valid(self, form):
+        result = super().form_valid(form)
+        self.object.advertisers.add(self.advertiser)
+        messages.success(
+            self.request,
+            _("Successfully invited %(user)s to %(advertiser)s")
+            % {"user": self.object.email, "advertiser": self.advertiser},
+        )
+        return result
+
+    def get_context_data(self, **kwargs):
+        context = super().get_context_data(**kwargs)
+        context.update({"advertiser": self.advertiser})
+        return context
+
+    def get_success_url(self):
+        return reverse(
+            "advertiser_users", kwargs={"advertiser_slug": self.advertiser.slug}
+        )
+
+
+class AdvertiserAuthorizedUsersRemoveView(
+    AdvertiserAccessMixin, UserPassesTestMixin, TemplateView
+):
+
+    """
+    Remove authorized users for an advertiser.
+
+    This doesn't remove or deactivate the user - just removes them from the advertiser.
```

Comment:
I wonder what our UX looks like for users with no advertisers or publishers. Might be something to consider here, once a user gets removed it might just look confusing and broken.

Reply:
They get a screen with "You don't have access to anything". https://github.com/readthedocs/ethical-ad-server/blob/main/adserver/templates/adserver/dashboard.html#L43-L45